### PR TITLE
Fix ProjectJobTokenScopes to use the correct endpoints

### LINKED ISSUE #3649: GroupJobTokenScopes should expect a project_id and a target_group_id
https://github.com/jdalrymple/gitbeaker/blob/77a20d94db8c03c70b372f8bffdb1594c18aa514/packages/core/src/resources/GroupJobTokenScopes.ts#L8

Groups do not have Job Token allow lists, only Projects have job token allow lists. 

The nuance is a project can either add a target group or a target project to its allow list.

https://docs.gitlab.com/ee/api/project_job_token_scopes.html#add-a-group-to-a-cicd-job-token-allowlist

### PR DESCRIPTION
**Pull request by jdalrymple**
Now correctly including the group allow lists

fixes #3649

### CODE REVIEW

File: packages/core/src/resources/ProjectJobTokenScopes.ts [hunk 1]
```diff
@@ -1,43 +1,115 @@
...
-    /* istanbul ignore next */
-    super('projects', options);
+  addToGroupsAllowList<E extends boolean = false>(
+    projectId: string | number,
+    targetGroupId: number,
+    options?: Sudo & ShowExpanded<E>,
+  ): Promise<GitlabAPIResponse<GroupAllowListSchema, C, E, void>> {
+    return RequestHelper.post<GroupAllowListSchema>()(
+      this,
+      endpoint`${projectId}/job_token_scope/groups_allowlist/${targetGroupId}`,
```
Comment (clario-masonb) [inline packages/core/src/resources/ProjectJobTokenScopes.ts:99]: For add, targetGroupId is a post body parameter, not a url parameter 
https://docs.gitlab.com/ee/api/project_job_token_scopes.html#add-a-group-to-a-cicd-job-token-allowlist

the target is only a url parameter for the remove endpoints

File: packages/core/src/resources/ProjectJobTokenScopes.ts [hunk 2]
```diff
@@ -1,43 +1,115 @@
...
 
   addToInboundAllowList<E extends boolean = false>(
     projectId: string | number,
     targetProjectId: string | number,
     options?: Sudo & ShowExpanded<E>,
-  ): Promise<GitlabAPIResponse<AllowListSchema, C, E, void>>;
+  ): Promise<GitlabAPIResponse<ProjectAllowListSchema, C, E, void>> {
+    return RequestHelper.post<ProjectAllowListSchema>()(
+      this,
+      endpoint`${projectId}/job_token_scope/allowlist/${targetProjectId}`,
```
Comment (clario-masonb) [inline packages/core/src/resources/ProjectJobTokenScopes.ts:64]: For add, targetProjectId is a post body parameter, not a url parameter 
https://docs.gitlab.com/ee/api/project_job_token_scopes.html#add-a-project-to-a-cicd-job-token-inbound-allowlist

the target is only a url parameter for the remove endpoints

**Review by clario-masonb — CHANGES_REQUESTED** [review]
Thanks much for working on the update! Just had a note for both of the "add" endpoints, after that I think we'll be solid.

**Comment by jdalrymple** [comment]
:rocket: PR was released in [`41.2.0`](https://github.com/jdalrymple/gitbeaker/releases/tag/41.2.0) :rocket:

**Pull request merged**